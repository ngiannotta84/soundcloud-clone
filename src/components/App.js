import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import "../styles/App.css";
import Navbar from "./Navbar";
import Home from "./Home";
import Login from "./Login";
import Search from "./Search";
import Upload from "./Upload";
import Profile from "./Profile";
import SignUp from "./SignUp";

const App = () => {
  return (
    <div className="App">
      <h2>SoundClone</h2>
      <Router>
        <Navbar />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/login" element={<Login />} />
          <Route path="/search" element={<Search />} />
          <Route path="/upload" element={<Upload />} />
          <Route path="/profile/:userName" element={<Profile />} />
          <Route path="/signup" element={<SignUp />} />
        </Routes>
      </Router>
    </div>
  );
};

export default App;
