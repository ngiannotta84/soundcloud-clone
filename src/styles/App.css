.app {
  text-align: center;
  background-color: var(--dark-grey);
  min-height: 100vh;
}

.body {
  margin: 0px;
  padding: 0px;
}

.app__routes {
  display: flex;
  justify-content: center;
}

.feed {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}