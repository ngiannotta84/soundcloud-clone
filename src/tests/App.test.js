import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

test("renders title", () => {
  render(<App />);
  const linkElement = screen.getByText(/soundclone/i);
  expect(linkElement).toBeInTheDocument();
});
