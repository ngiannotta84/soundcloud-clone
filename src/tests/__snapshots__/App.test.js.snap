// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="navbar"
    >
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links-item"
        >
          <h1>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              SOUNDCLONE
            </a>
          </h1>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="navbar-links-item"
        >
          <input
            type="text"
          />
          <button
            type="button"
          >
            search
          </button>
        </li>
        <li
          class="navbar-links-item"
        >
          <a
            class=""
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
    <div>
      <div
        class="album"
      >
        <div
          class="album__info-container"
        >
          <img
            alt="album1 cover art"
            class="album__cover-art"
            src="art url 1"
          />
          <div>
            <h2
              class="album__artist-name"
            >
              artist name 1
            </h2>
            <h3
              class="album__album-name"
            >
              album1
            </h3>
          </div>
        </div>
        <div>
          <div
            class="album__song"
          >
            <h4
              class="album__song__song-name"
            >
              song1
            </h4>
            <div>
              <button
                class="album__song__add-song"
                type="button"
              >
                Add to queue
              </button>
              <button
                class="album__song__add-song"
                type="button"
              >
                Play next
              </button>
            </div>
          </div>
          <div
            class="album__song"
          >
            <h4
              class="album__song__song-name"
            >
              song2
            </h4>
            <div>
              <button
                class="album__song__add-song"
                type="button"
              >
                Add to queue
              </button>
              <button
                class="album__song__add-song"
                type="button"
              >
                Play next
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="album"
      >
        <div
          class="album__info-container"
        >
          <img
            alt="album2 cover art"
            class="album__cover-art"
            src="art url 2"
          />
          <div>
            <h2
              class="album__artist-name"
            >
              artist name 2
            </h2>
            <h3
              class="album__album-name"
            >
              album2
            </h3>
          </div>
        </div>
        <div>
          <div
            class="album__song"
          >
            <h4
              class="album__song__song-name"
            >
              song4
            </h4>
            <div>
              <button
                class="album__song__add-song"
                type="button"
              >
                Add to queue
              </button>
              <button
                class="album__song__add-song"
                type="button"
              >
                Play next
              </button>
            </div>
          </div>
          <div
            class="album__song"
          >
            <h4
              class="album__song__song-name"
            >
              song3
            </h4>
            <div>
              <button
                class="album__song__add-song"
                type="button"
              >
                Add to queue
              </button>
              <button
                class="album__song__add-song"
                type="button"
              >
                Play next
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <audio
      data-testid="audio"
    />
    <div
      class="music-player"
    >
      <div
        class="music-player__left"
      >
        <img
          alt="undefined cover art"
          class="music-player__cover-art"
          src="music.png"
        />
        <div
          class="music-player__info"
        >
          <h2 />
          <h3 />
        </div>
      </div>
      <div
        class="music-player__center"
      >
        <input
          class="music-player__progress-bar"
          data-testid="progressBar"
          max="0"
          min="0"
          name="time"
          type="range"
          value="0"
        />
        <div
          class="music-player__center__under"
        >
          <p>
            0:00
          </p>
          <div>
            <button
              class="music-player__skip-backwards"
              type="button"
            >
              <img
                alt="skip backwards"
                src="skip.png"
              />
            </button>
            <button
              class="music-player__play-pause"
              type="button"
            >
              <img
                alt="pause"
                src="play.png"
              />
            </button>
            <button
              class="music-player__skip-forwards"
              type="button"
            >
              <img
                alt="skip forwards"
                src="skip.png"
              />
            </button>
          </div>
          <p>
            0:00
          </p>
        </div>
      </div>
      <div
        class="music-player__right"
      >
        <div
          class="music-player__volume"
        >
          <button
            type="button"
          >
            <img
              alt="volume"
              src="volume-medium.png"
            />
          </button>
        </div>
        <div
          class="music-player__playlist"
        >
          <button
            type="button"
          >
            <img
              alt="playlist"
              src="menu.png"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
